pkg/controller/stacks/job.go

```diff
@@ -205,12 +206,18 @@ func (r *StackReconciler) getDefaultContainerImage(run *console.StackRunFragment
 	return fmt.Sprintf("%s:%s-%s-%s", image, defaultImageTag, strings.ToLower(string(run.Type)), version)
 }
 
-func (r *StackReconciler) getDefaultContainerArgs(runID string) []string {
-	return []string{
+func (r *StackReconciler) getDefaultContainerArgs(run *console.StackRunFragment) []string {
+	args := []string{
 		fmt.Sprintf("--console-url=%s", r.ConsoleURL),
 		fmt.Sprintf("--console-token=%s", r.DeployToken),
-		fmt.Sprintf("--stack-run-id=%s", runID),
+		fmt.Sprintf("--stack-run-id=%s", run.ID),
 	}
+
+	if run.Workdir != nil && len(*run.Workdir) > 0 {
```

Comment:
We shouldn't do this like i said, it's actually intended to have the workdir be a subdirectory of the overall dir

Reply:
Moved to harness logic